spec/integration/driver_spec.rb

```diff
@@ -439,6 +439,84 @@ def create_screenshot(file, *args)
         @session.visit('/poltergeist/redirect_to_headers')
         expect(@driver.body).not_to include('X_CUSTOM_HEADER: 1')
       end
+
+      context 'multiple windows' do
+        it 'persists headers across popup windows' do
+          @driver.headers = {
+            'Cookie' => 'foo=bar',
+            'Host' => 'foo.com',
+            'User-Agent' => 'foo'
+          }
+          @session.visit('/poltergeist/popup_headers')
+          @session.click_link 'pop up'
+          @session.switch_to_window @session.windows.last
+          expect(@driver.body).to include('USER_AGENT: foo')
+          expect(@driver.body).to include('COOKIE: foo=bar')
+          expect(@driver.body).to include('HOST: foo.com')
+        end
+
+        it 'sets headers in existing windows' do
+          @session.open_new_window
```

Comment:
`open_new_window` returns the new window object - use that rather than depending on the order of `@sessions.windows`